extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisAPIProducer.java

```diff
@@ -0,0 +1,126 @@
+package io.quarkus.redis.client.runtime;
+
+import java.net.InetSocketAddress;
+import java.util.Set;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.quarkus.redis.client.SyncRedisAPI;
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.vertx.core.Vertx;
+import io.vertx.redis.client.Redis;
+import io.vertx.redis.client.RedisAPI;
+import io.vertx.redis.client.RedisClientType;
+import io.vertx.redis.client.RedisOptions;
+
+@ApplicationScoped
+class RedisAPIProducer {
+    private static final char AT = '@';
+    private static final char COLON = ':';
+    private static final char SLASH = '/';
+    private static final String REDIS_SCHEME = "redis://";
+    private static final String REDIS_SSL_SCHEME = "rediss://";
+
+    private long timeout = 10;
+
+    private final RedisConfig config;
+
+    private final Redis redisClient;
+
+    private final RedisAPI redisAPI;
+
+    private final SyncRedisAPI syncRedisAPI;
+
+    private final io.vertx.mutiny.redis.client.Redis mutinyRedisClient;
+
+    private final io.vertx.mutiny.redis.client.RedisAPI mutinyRedisAPI;
+
+    public RedisAPIProducer(RedisConfig config, Vertx vertx) {
+        this.config = config;
+        RedisOptions options = new RedisOptions();
+        options.setType(config.clientType);
+
+        if (RedisClientType.STANDALONE == config.clientType) {
+            if (config.hosts.isPresent() && config.hosts.get().size() > 1) {
+                throw new ConfigurationException("Multiple hosts supplied for non clustered configuration");
+            }
+        }
+
+        if (config.hosts.isPresent()) {
+            Set<InetSocketAddress> hosts = config.hosts.get();
+            for (InetSocketAddress host : hosts) {
+                String connectionString = buildConnectionString(host);
+                options.addConnectionString(connectionString);
+            }
+        } else {
+            InetSocketAddress defaultRedisAddress = new InetSocketAddress("localhost", 6379);
```

Comment:
Make sure this is documented as it does not use the default value attribute.

Reply:
It should be automatically documented with this default value https://github.com/quarkusio/quarkus/pull/9370/files#diff-61bc41cc0a7b78f75d4088c283c29ffbR25